gradle-recommended-product-dependencies/src/main/java/com/palantir/gradle/dist/CompileRecommendedProductDependencies.java

```diff
@@ -17,36 +17,46 @@
 package com.palantir.gradle.dist;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import java.io.File;
 import java.io.IOException;
+import javax.inject.Inject;
 import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
 import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFileProperty;
 import org.gradle.api.provider.SetProperty;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
 
 public abstract class CompileRecommendedProductDependencies extends DefaultTask {
 
+
+    public static final String RESOURCE_PATH =
+            RecommendedProductDependencies.SLS_RECOMMENDED_PRODUCT_DEPS_KEY + "/product-dependencies.json";
     static final ObjectMapper MAPPER = new ObjectMapper();
 
     @Input
     abstract SetProperty<ProductDependency> getRecommendedProductDependencies();
 
+    @OutputFile
+    abstract RegularFileProperty getOutputFile();
+
     @OutputDirectory
-    abstract DirectoryProperty getOutputDir();
+    abstract DirectoryProperty getOutputDirectory();
```

Comment:
We should explain in a comment why this needs to be an `@OutputDirectory` - it confused me at first. We should also have a test that a task dependency is set up properly between the source jar and this task. Probably should have had them in [the previous PR](https://github.palantir.build/gradle-plugins/dev-env/pull/3546), but will do to have them here.